test/test_stdout.rb

```diff
@@ -18,4 +18,9 @@ def test_default_list_output
       assert_match(/.*#{spec_key}.*/, @stdout.string)
     end
   end
+
+  def test_ubuntu_host_generation_output
```

Comment:
> I just chose the Ubuntu string because I knew it would hit the warning that gets written to stderr, should I just replace it with a simpler token? Something like centos6-64ma?

The way I think about testing this is...

- The warning isn't platform-specific, so I would not want to make it seem like it is.

   So I would remove any mention of the platforms from the test name. 

- It also has nothing to do with the number of hosts or their roles, so I would choose the simplest input that would work.

   So I would simplify `'ubuntu1404-64default.mdcal-ubuntu1404-64af'` to `'ubuntu1404-64'`.